internal/services/synapse/synapse_pipeline_resource_test.go

```diff
@@ -0,0 +1,357 @@
+  synapse_workspace_id = azurerm_synapse_workspace.test.id
+
+  parameters = {
+    test = "testparameter"
+  }
+
+  variables = {
+    foo = "test1"
+    bar = "test2"
+  }
+
+  depends_on = [azurerm_synapse_firewall_rule.test]
+}
+`, template, data.RandomInteger)
+}
+
+func (r PipelineResource) update1(data acceptance.TestData) string {
+	template := r.template(data)
+	return fmt.Sprintf(`
+	%s
+
+resource "azurerm_synapse_pipeline" "test" {
+  name                 = "acctest%d"
+  synapse_workspace_id = azurerm_synapse_workspace.test.id
+  annotations          = ["test1", "test2", "test3"]
+  description          = "test description"
+
+  parameters = {
+    test = "testparameter"
+  }
+
+  variables = {
+    foo = "test1"
+    bar = "test2"
+  }
+
+  depends_on = [azurerm_synapse_firewall_rule.test]
+}
+`, template, data.RandomInteger)
+}
+
+func (r PipelineResource) update2(data acceptance.TestData) string {
+	template := r.template(data)
+	return fmt.Sprintf(`
+	%s
+
+resource "azurerm_synapse_pipeline" "test" {
+  name                 = "acctest%d"
+  synapse_workspace_id = azurerm_synapse_workspace.test.id
+  annotations          = ["test1", "test2"]
+  description          = "test description2"
+
+  parameters = {
+    test  = "testparameter"
+    test2 = "testparameter2"
+  }
+
+  variables = {
+    foo = "test1"
+    bar = "test2"
+    baz = "test3"
+  }
+
+  activities_json = <<JSON
+[
+  {
+    "name": "Append variable1",
+    "type": "AppendVariable",
+    "dependsOn": [],
+    "userProperties": [],
+    "typeProperties": {
+      "variableName": "foo",
+      "value": "something"
+    }
+  }
+]
+JSON
+
+  depends_on = [azurerm_synapse_firewall_rule.test]
+}
+`, template, data.RandomInteger)
+}
+
+func (r PipelineResource) activities(data acceptance.TestData) string {
+	template := r.template(data)
+	return fmt.Sprintf(`
+	%s
+
+resource "azurerm_synapse_pipeline" "test" {
+  name                 = "acctest%d"
+  synapse_workspace_id = azurerm_synapse_workspace.test.id
+  variables = {
+    "bob" = "item1"
+  }
+  activities_json = <<JSON
+[
+  {
+    "name": "Append variable1",
+    "type": "AppendVariable",
+    "dependsOn": [],
+    "userProperties": [],
+    "typeProperties": {
+      "variableName": "bob",
+      "value": "something"
+    }
+  }
+]
+JSON
+
+  depends_on = [azurerm_synapse_firewall_rule.test]
+}
+`, template, data.RandomInteger)
+}
+
+func (r PipelineResource) activitiesUpdated(data acceptance.TestData) string {
+	template := r.template(data)
+	return fmt.Sprintf(`
+	%s
+
+resource "azurerm_synapse_pipeline" "test" {
+  name                 = "acctest%d"
+  synapse_workspace_id = azurerm_synapse_workspace.test.id
+  variables = {
+    "bob" = "item1"
+  }
+  activities_json = <<JSON
+[
+  {
+    "name": "Append variable1",
+    "type": "AppendVariable",
+    "dependsOn": [],
+    "userProperties": [],
+    "typeProperties": {
+      "variableName": "bob",
+      "value": "something"
+    }
```

Comment:
Can we add multiple activities here so that we can check the order of them are reserved?

Reply:
Sure. Added.